### PR TITLE
DAOS-10986 pool: Add forgotten s_mutex unlock

### PR DESCRIPTION
**Pull request by liw**
The ds_pool_create_handler function should unlock ds_rsvc.s_mutex if rc
is not zero when ending the RDB TX.

Signed-off-by: Li Wei <wei.g.li@intel.com>

### CODE REVIEW

**Comment by liw** [comment]
POOL14: DAOS-10301

**Review by NiuYawei — APPROVED** [review]

File: src/pool/srv_pool.c [hunk 1]
```diff
@@ -2170,7 +2170,7 @@ ds_pool_create_handler(crt_rpc_t *rpc)
 	ABT_rwlock_unlock(svc->ps_lock);
 	rdb_tx_end(&tx);
 	if (rc != 0)
-		D_GOTO(out_svc, rc);
+		D_GOTO(out_mutex, rc);
```
Comment (kccain) [inline src/pool/srv_pool.c:2173]: Is the D_GOTO(out_svc, rc) on line 2189 in the conditional below also needing a change to D_GOTO(out_mutex, rc) ?

Comment (liw) [reply src/pool/srv_pool.c:2173]: Good catch. I don't know how I missed that one.

**Review by kccain — APPROVED** [review]

**Review by fengchunsong — APPROVED** [review]
LGTM

**Comment by liw** [comment]
REBUILD4 EC: DAOS-10832

**Review by liuxuezhao — APPROVED** [review]

**Pull request merged**